core/src/main/java/org/commonjava/maven/ext/core/impl/RelocationManipulator.java

```diff
@@ -203,9 +289,298 @@ private void updateDependencyExclusion( ProjectVersionRef depPvr, ProjectVersion
+
+            for ( final Map.Entry<ConfigurationContainer, String> entry : configs.entrySet() )
+            {
+                try
+                {
+                    final Document doc = galleyWrapper.parseXml( entry.getValue() );
+                    final XPath xPath = XPathFactory.newInstance().newXPath();
+
+                    for ( String expression : expressions )
+                    {
+                        final NodeList children = (NodeList) xPath.evaluate( expression, doc, XPathConstants.NODESET );
+                        final PluginReference ref = updatePluginGAVs( entry, children );
+
+                        if ( ref != null )
+                        {
+                            refs.add( ref );
+                        }
+                    }
+                }
+                catch ( final GalleyMavenXMLException e )
+                {
+                    throw new ManipulationException( "Unable to parse config for plugin {} in {}", plugin.getId(), project.getKey(), e );
+                }
+                catch ( final  XPathExpressionException e )
+                {
+                    throw new ManipulationException( "Invalid XPath expression for plugin {} in {}", plugin.getId(), project.getKey(), e );
+                }
+            }
+        }
+
+        return refs;
+    }
+
+    private Map<String, Plugin> getPluginMap( final ModelBase base )
+    {
+        final BuildBase build;
+        if ( base instanceof Model )
+        {
+            build = ( (Model) base ).getBuild();
+        }
+        else
+        {
+            build = ( (Profile) base ).getBuild();
+        }
+
+        if ( build == null )
+        {
+            return Collections.emptyMap();
+        }
+
+        final Map<String, Plugin> result = build.getPluginsAsMap();
+        if ( result == null )
+        {
+            return Collections.emptyMap();
+        }
+
+        return result;
+    }
+
+    private static Map<String, Plugin> getManagedPluginMap( final ModelBase base )
+    {
+        if ( base instanceof Model )
+        {
+            final Build build = ( (Model) base ).getBuild();
+            if ( build == null )
+            {
+                return Collections.emptyMap();
+            }
+
+            final PluginManagement pm = build.getPluginManagement();
+            if ( pm == null )
+            {
+                return Collections.emptyMap();
+            }
+
+            final Map<String, Plugin> result = pm.getPluginsAsMap();
+            if ( result == null )
+            {
+                return Collections.emptyMap();
+            }
+
+            return result;
+        }
+
+        return Collections.emptyMap();
+    }
+
+    /**
+     * Go through the plugin / plugin-execution configurations and find references to the <code>skip</code> parameter for the given Maven plugin
+     * instance.
```

Comment:
I think this doc isn't right ?

Reply:
Right. This was taken from other code, so maybe there was some reason for keying the map like this. I am not finding a reference to the key, in which case, we may be able to change `Map<ConfigurationContainer, String>` to `List<String>` as well.